File: Feed/src/main/java/com/j/jface/feed/BootCompletedReceiver.java
package com.j.jface.feed;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootCompletedReceiver extends BroadcastReceiver
{
  @Override public void onReceive(final Context context, final Intent intent)
  {
    final Intent i = new Intent(context, GeofenceTransitionReceiverService.class);
    i.setAction(Intent.ACTION_BOOT_COMPLETED);
    context.startService(i);
  }
}

File: Feed/src/main/java/com/j/jface/feed/BootCompletedReceiver.kt
package com.j.jface.feed

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootCompletedReceiver : BroadcastReceiver()
{
  override fun onReceive(context : Context, intent : Intent)
  {
    if (intent.action != Intent.ACTION_BOOT_COMPLETED) return // Security suggestion from linter : spoofing countermeasure

    val i = Intent(context, GeofenceTransitionReceiver::class.java)
    i.action = Intent.ACTION_BOOT_COMPLETED
    GeofenceTransitionReceiver().onReceive(context, i)
  }
}
